src/query/api/v1/handler/namespace/common.go

```diff
@@ -190,6 +191,16 @@ func validateNamespaceAggregationOptions(mds []namespace.Metadata) error {
 				resolutionRetentionMap[key] = true
 			}
 		}
+
```

Comment:
Is this API the only way to create a new NS?

Reply:
Yeah I believe this is a common path for all creations. Though Artem's suggestion of putting in the Validate probably even better.